File: python/arg-parse.py
#!/usr/bin/python
#
# author: yin-jianhong@163.com
#
# ref: https://docs.python.org/3/library/argparse.html
#
# coding=utf-8

import sys
import string
import argparse

parser = argparse.ArgumentParser(description="some information here")

parser.add_argument("echo", help="echo the string you use here")

# type=
parser.add_argument("square", help="display a square of a given number",
                type=int)

# action=
parser.add_argument("-e", "--enable", help="enable xxx",
                action="store_true")

# type & choices
parser.add_argument("-V", "--Verbosity", type=int, choices=[0, 1, 2],
                help="increase output verbosity")

# action='count' default=0
parser.add_argument("-v", "--verbosity", action="count", default=0,
                help="increase output verbosity")

# nargs=
parser.add_argument('--foo', type=int, nargs='?', action='append',
                help='an integer for the accumulator')
parser.add_argument('--bar', type=int, nargs='+', action='append',
                help='an integer for the accumulator')
parser.add_argument('--boo', type=list, nargs='*', action='append',
                help='an integer for the accumulator')

args = parser.parse_args()

parser.print_help()

if args.enable:
    print "enable = %s" % args.enable
if args.Verbosity:
    print "Verbosity = %d" % args.Verbosity
if args.verbosity:
    print "verbosity = %d" % args.verbosity
if args.foo:
    print "foo = ", args.foo
if args.bar:
    print "bar = ", args.bar
if args.boo:
    print "boo = ", args.boo

File: R/argparse.R
#!/usr/bin/env Rscript

# maybe u need call install.package("argparse") first
suppressPackageStartupMessages(library("argparse"))

# create parser object
parser <- ArgumentParser(description='*Process some info/description*')

parser$add_argument("-v", "--verbose", action="store_true", default=FALSE,
    help="output verbose detail info")
parser$add_argument("-q", "--quietly", action="store_false",
    dest="quiet", help="suppress detail output")
parser$add_argument("-c", "--count", type="integer", default=5,
    help="Number of random normals to generate [default %(default)s]",
    metavar="number")
parser$add_argument("--sd", default=1, type="double",
    metavar="standard deviation",
    help="abcde [default %(default)s]")
parser$add_argument('--sum', dest='accumulate', action='store_const',
    const='sum', default='max', help='sum the integers (default: find the max)')

parser$add_argument('start', metavar='N', type="integer", nargs='+',
    help='an integer for the accumulator')
parser$add_argument('end', metavar='N', type="integer", nargs='+',
    help='an integer for the accumulator')

# don't need add -h/--help option, will generate
args <- parser$parse_args()

# print some progress messages to stderr if "quietly" wasn't requested
if ( args$verbose ) {
    write("writing some verbose output to standard error...\n", stderr())
}

# do some operations based on user input
cat(args$start:args$end)
cat("\n")
